### ISSUE TITLE
listener passed to SocketIO.on() not being called

### ISSUE THREAD

**Issue by jweidner-mbible** [infitio/flutter_socket_io#131]
D/Adhara was logging Socket triggered::event-name but our listener was not getting called nor was any error being logged.  I had declared our listener as:
```dart
void _handleMessage( dynamic data ) {
    print( '_handleMessage received $data' );
}
```
and subscribed to the events using
```dart
  socket.on( 'eventName', _handleMessage );
```
In debugging this situation, I noticed that when _handleData in socket.dart would try to call our listener using Function.apply, the arguments List contained two elements instead of one.  So to get this to work, I needed to change the signature of our listener to allow for a second argument.  But since the listener signature required to be passed to socket.on() was only expecting a single argument, I needed to make the second argument optional.  Changing our _handleMessage signature as follows, fixed our issue:
```dart
void _onMessage(dynamic channelName, [dynamic data]) {
    print( '_handleMessage received $data' );
}
```

**Comment by deepashree-ballodi** [comment]
Is this issue fixed? I am facing same issue.

**Comment by idevchandra** [comment]
@tiholic @jweidner-mbible @deepashree-ballodi @alexHimik : Guys, please please respond to the issues. There are pending ones. We invested project schedules and time in using this library. Please address the issues.

**Comment by tiholic** [comment]
@idevchandra @deepashree-ballodi @jweidner-mbible 

You can now use Streams to listen to events from newly released v1.0.0.
Please try. Re-open this issue or create a new ones in case it doesn't work.
